ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake

```diff
@@ -24,5 +26,30 @@ file(GLOB_RECURSE _source_files FOLLOW_SYMLINKS
 )
 if(_source_files)
   message(STATUS "Added test 'cppcheck' to perform static code analysis on C / C++ code")
-  ament_cppcheck()
+
+  # Get include paths for added targets
+  set(_all_include_dirs "")
+  # BUILDSYSTEM_TARGETS only supported in CMake >= 3.7
+  if(NOT CMAKE_VERSION VERSION_LESS "3.7.0")
+    get_directory_property(_build_targets DIRECTORY ${CMAKE_SOURCE_DIR} BUILDSYSTEM_TARGETS)
+    foreach(_target ${_build_targets})
+      get_property(_include_dirs
+        TARGET ${_target}
+        PROPERTY INCLUDE_DIRECTORIES
+      )
+
+      # Only append include directories that are from the package being tested
```

Comment:
I'm seeing a similar cppcheck complaint in `rmw_connext` and `rmw_opensplice` where they use a macro `RMW_CHECK_TYPE_IDENTIFIERS_MATCH` that is defined `rmw`.

https://ci.ros2.org/view/nightly/job/nightly_osx_extra_rmw_release/313/testReport/junit/rmw_opensplice_cpp/cppcheck/error__unknownMacro__src_rmw_wait_cpp_80_/

https://ci.ros2.org/job/nightly_osx_debug/1175/testReport/junit/rmw_connext_shared_cpp/cppcheck/error__unknownMacro___Users_osrf_jenkins_agent_workspace_nightly_osx_debug_ws_src_ros2_rmw_connext_rmw_connext_shared_cpp_include_rmw_connext_shared_cpp_wait_hpp_51_/

Any way to tell cppcheck where to look for macros without checking external files?

Reply:
@sloretz See #125.

You can pass cppcheck one or more include directories that contain the macros it is trying to resolve. 
I'm not satisfied with the solution, but I think anything better would require changes in `cppcheck` itself.

Reply:
thanks, I missed that PR